### PR TITLE
Feature/create article controller and dto

### PR DESCRIPTION
**Pull request by Olga232**

### CODE REVIEW

File: src/main/java/baza/trainee/domain/dto/ArticleDto.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+public record ArticleDto(
+        String id,
+        String title,
+        String description,
+        String content,
+        Set<String> images,
+        LocalDate created,
+        LocalDate updated
+) {
+}
```
Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/domain/dto/ArticleDto.java]: Should be validation of DTO fields.

File: src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    /**
+     * Finds an existing article by given id.
+     *
+     * @param id id to get an existing article.
+     * @return {@link ArticleDto} object containing an existing article with its full content.
+     */
+    @Override
+    public ArticleDto findById(final String id) {
+        return null;
+    }
```
Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java:18]: ArticleService should return Article or ArticleDto?

File: src/main/java/baza/trainee/controller/ArticleController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * Finds an existing article by given id.
+     *
+     * @param articleId id to get an existing article.
+     * @return {@link ArticleDto} object containing an existing article with its full content.
+     */
+    @GetMapping(value = "/{articleId}", produces = "application/json")
+    public ArticleDto findById(@PathVariable(name = "articleId") final String articleId) {
+        return articleService.findById(articleId);
+    }
```
Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/controller/ArticleController.java:36]: If service will be return Article, we need additional conversion to DTO.

File: src/test/java/baza/trainee/controller/ArticleControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    LoggingService loggingService;
+
+    @Test
+    @SneakyThrows
+    void findById() {
+        //GIVEN
+        final var articleDto = new ArticleDto("testId", "testTitle", "testDescription",
+                "testContent", new HashSet<>(), LocalDate.now(), null);
+
+        when(articleService.findById(articleDto.id())).thenReturn(articleDto);
```
Comment (Anatolii-Omelchenko) [inline src/test/java/baza/trainee/controller/ArticleControllerTest.java]:
```suggestion
        //WHEN
        when(articleService.findById(articleDto.id())).thenReturn(articleDto);
        
        //THEN
```

File: src/test/java/baza/trainee/controller/ArticleControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+        //WHEN
+        final var result = mockMvc.perform(get("/api/article/{articleId}", articleDto.id()));
+
+        //THEN
+        result
+                .andDo(print())
+                .andExpect(status().isOk());
+    }
```
Comment (Anatolii-Omelchenko) [inline src/test/java/baza/trainee/controller/ArticleControllerTest.java]:
```suggestion
        //THEN
        mockMvc.perform(get("/api/article/{articleId}", articleDto.id()))
                .andDo(print())
                .andExpect(status().isOk());
    }
```
u can do it in one line (without additional variable).

File: src/main/java/baza/trainee/domain/model/Article.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+ * The domain model of the static content of a web page.
+ *
+ * @author Olha Ryzhkova
+ * @version 1.0
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor(access = AccessLevel.PROTECTED)
+@Document
+public class Article {
```
Comment (Anatolii-Omelchenko) [inline src/main/java/baza/trainee/domain/model/Article.java:31]: Also we can testing Models and DTOs (Validation).

File: pom.xml [hunk 7]
```diff
@@ -35,22 +35,15 @@
             <version>1.18.28</version>
         </dependency>
         <dependency>
-            <groupId>org.springframework.boot</groupId>
```
Comment (malyshevhen) [inline pom.xml:38]: Do not delete this dependency

File: src/main/java/baza/trainee/controller/ArticleController.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Spring MVC REST controller serving article operations for non-admin users.
+ *
+ * @author Olha Ryzhkova
+ * @version 1.0
+ */
+@RestController
+@RequestMapping("/api/article")
+@RequiredArgsConstructor
+public class ArticleController {
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/controller/ArticleController.java:20]: We also need a non-secure endpoint for retrieving the page of articles.

File: pom.xml [hunk 9]
```diff
@@ -35,22 +35,15 @@
...
+            <version>0.8.6</version>
         </dependency>
-
         <dependency>
             <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-web</artifactId>
-        </dependency>
-
-        <dependency>
-            <groupId>org.projectlombok</groupId>
```
Comment (malyshevhen) [inline pom.xml:49]: And this to

File: src/main/java/baza/trainee/controller/ArticleController.java (hunk 8 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/controller/ArticleController.java:20]: I think it meant method with Pageable

File: src/test/java/baza/trainee/controller/ArticleControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import java.time.LocalDate;
+import java.util.HashSet;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(ArticleController.class)
+class ArticleControllerTest {
```
Comment (Anatolii-Omelchenko) [inline src/test/java/baza/trainee/controller/ArticleControllerTest.java:21]: u can add negative cases

File: src/test/java/baza/trainee/domain/dto/ArticleDtoTest.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+                "testContent", new HashSet<>(), LocalDate.now(),null);
+        NOT_VALID_ARTICLE_DTO = new ArticleDto("testId", "", " ", null,
+                new HashSet<>(), LocalDate.now(), null);
+    }
+
+    @BeforeEach
+    void setUp() {
+        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+        validator = factory.getValidator();
+    }
```
Comment (Anatolii-Omelchenko) [inline src/test/java/baza/trainee/domain/dto/ArticleDtoTest.java]:
```suggestion
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class ArticleDtoTest {

    private Validator validator;
    private ArticleDto validArticleDto;
    private ArticleDto notValidArticleDto;

    @BeforeAll
    void init() {
        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
        validator = factory.getValidator();

        validArticleDto = new ArticleDto("testId", "testTitle", "testDescription",
                "testContent", new HashSet<>(), LocalDate.now(), null);
        notValidArticleDto = new ArticleDto("testId", "", " ", null,
                new HashSet<>(), LocalDate.now(), null);
    }
```

File: src/test/java/baza/trainee/controller/ArticleControllerTest.java (hunk 10 above)
Comment (Olga232) [reply src/test/java/baza/trainee/controller/ArticleControllerTest.java:21]: added

File: src/test/java/baza/trainee/domain/dto/ArticleDtoTest.java (hunk 11 above)
Comment (Olga232) [reply src/test/java/baza/trainee/domain/dto/ArticleDtoTest.java]: updated

File: src/main/java/baza/trainee/controller/ArticleController.java (hunk 8 above)
Comment (Olga232) [reply src/main/java/baza/trainee/controller/ArticleController.java:20]: For now, we don't have a business case for using the page of articles. I suggest adding this endpoint later if necessary.

**Review by malyshevhen — APPROVED** [review]

**Review by malyshevhen — APPROVED** [review]

**Review by Anatolii-Omelchenko — APPROVED** [review]

**Pull request merged**